### PR TITLE
Cleanup top level POM dependencies

### PR DESCRIPTION
**Pull request by apc999**
### What changes are proposed in this pull request?

Cleanup top level POM dependencies

- Move dependency version definition to <dependencyManagement> section
- Remove unused properties
- Sort properties and dependencies alphabetically
- Remove unused profile

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -311,26 +337,26 @@
...
         <artifactId>commons-io</artifactId>
         <version>2.7</version>
       </dependency>
       <dependency>
-        <groupId>commons-logging</groupId>
-        <artifactId>commons-logging</artifactId>
-        <version>1.2</version>
+        <groupId>io.atomix.catalyst</groupId>
+        <artifactId>catalyst-transport</artifactId>
+        <version>${catalyst.version}</version>
```
Comment (jiacheliu3) [inline pom.xml:353]: for some of the dependencies we use a param like this, and for others we directly put a version string here. Do you think it makes sense to unify the two flavors?

File: pom.xml [hunk 2]
```diff
@@ -1462,16 +1496,6 @@
       </build>
     </profile>
 
-    <!-- profile that Alluxio developers should use -->
-    <profile>
```
Comment (jiacheliu3) [inline pom.xml:1466]: is this unused?

**Review by jiacheliu3 — APPROVED** [review]
The changes themselves LGTM, with a few minor comments. Thanks for the work!

File: pom.xml (hunk 1 above)
Comment (apc999) [reply pom.xml:353]: I do see the reasons to keep both flavors:
- with important deps, with set a param to indicate the version, like  `java.version`. Also sometimes we have deps of different artifacts need to share the same version, like `grpc.version` needs to be consistent across `grpc-api` and `grpc-core`.
- many other deps do not deserve a param :)

File: pom.xml [hunk 3]
```diff
@@ -122,62 +122,63 @@
...
+    <slf4j.version>1.7.30</slf4j.version>
     <stateless4j.version>2.6.0</stateless4j.version>
-    <surefire.forkCount>2</surefire.forkCount>
-    <surefire.useSystemClassLoader>true</surefire.useSystemClassLoader>
     <surefire.excludesFile>slow-tests</surefire.excludesFile>
+    <surefire.forkCount>2</surefire.forkCount>
     <surefire.rerunFailingTestsCount>1</surefire.rerunFailingTestsCount>
+    <surefire.useSystemClassLoader>true</surefire.useSystemClassLoader>
     <test.output.redirect>true</test.output.redirect>
     <update.check.auth.string>YWxsdXhpbzp0YWNoeW9u</update.check.auth.string>
```
Comment (LuQQiu) [inline pom.xml:178]: @Xenorith do you know what is this auth string? looks like my gitguadian warn me about it

Comment (Xenorith) [reply pom.xml:178]: it's a string used to verify the requests received by the update check server. we don't want to try to handle any request from other sources. there is no problem with exposing it in the source code

**Pull request merged**